kong/plugins/acl/handler.lua

```diff
@@ -81,7 +81,9 @@ function ACLHandler:access(conf)
     return responses.send_HTTP_FORBIDDEN("You cannot consume this service")
   end
 
-  set_header(constants.HEADERS.CONSUMER_GROUPS, to_be_blocked)
+  if not conf.hide_groups_header then
+    set_header(constants.HEADERS.CONSUMER_GROUPS, to_be_blocked)
+  end
```

Comment:
In the case when we're not setting the header, we should set a dummy value in the cache. We're trying to save bandwidth in the request, but we should also reduce Kong memory pressure.

so above do:
```lua
    if to_be_blocked == false then
       -- we're allowed, so go and convert 'false' to the header value, if we need it.
       -- if we don't need it, set a dummy value to save memory for potentially very long strings
       to_be_blocked = conf.hide_groups_header and "" or table_concat(consumer_groups, ", ")
     end
```

Reply:
This seems nice as a further improvement. Since this has already been merged to master branch would the proper approach be to submit a new PR with this additional enhancement? Didn't know you could do  *true* and "" or *table of consumers_groups* would cause to_be_blocked  == "".